### PR TITLE
implement flat_map() in php

### PR DESCRIPTION
**Pull request by nopolabs**
fixes https://github.com/lstrojny/functional-php/issues/49

flat_map provides a single level of flattening after a map application, per http://martinfowler.com/articles/collection-pipeline/flat-map.html

only implements flat_map in php
does not implement flat_map in native C

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -376,33 +376,6 @@ F\with($value, function($value) {
 });
 ```
 
-### Functional\sort()
```
Comment (lstrojny) [inline README.md]: Why is sort() removed?

**Comment by nopolabs** [comment]
Deleting sort() from README.md was a mistake. I have corrected it.

**Comment by lstrojny** [comment]
Can you check the failing tests?

**Comment by nopolabs** [comment]
tests passing

**Comment by lstrojny** [comment]
Cool! Looking good now, can you squash so I can merge? Thank you!

**Comment by nopolabs** [comment]
squashed

**Comment by lstrojny** [comment]
Thank you!

**Comment by nopolabs** [comment]
:)

On Tue, May 12, 2015 at 7:10 AM, Lars Strojny <notifications@github.com>
wrote:

> Thank you!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lstrojny/functional-php/pull/53#issuecomment-101296244>
> .
>



-- 
Dan Revel
danrevel@gmail.com

**Pull request merged**